### PR TITLE
[MAT-6046] Replace TGT/ST Usage with Basic Authentication

### PR DESCRIPTION
**Pull request by jkotanchik-SB**
## Terminology Service PR

Jira Ticket: [MAT-6046](https://jira.cms.gov/browse/MAT-6046)
(Optional) Related Tickets:

### Summary

VSAC has deprecated the Ticket Granting Ticket (TGT) / Service Ticket (ST) Authentication flow in favor of using Basic Authentication with the submitter's UMLS ApiKey.

Remove TGT and TGT expiry date from UmlsUser model.

Remove methods and calls for generating TGT/STs. 
Remove method parameters expecting TGT/STs.
Remove references to TGT/STs from URL templates.

Remove references to TGT/STs from unit tests.
Remove unit tests specifically targeting TGT/ST workflow.

Add Basic Authn to VSAC calls passing user's apiKey.
Update test case to mock basic authn header.

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [x] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [x] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -32,8 +32,8 @@ spring:
     max-in-memory-size: 24MB
   data:
     mongodb:
-      uri: ${MONGO_URI:mongodb://${DBUSER:root}:${DBPASS:E5press0}@localhost:27017/terminology}?authSource=admin&maxPoolSize=50&connectTimeoutMS=2000&serverSelectionTimeoutMS=2000
+      uri: mongodb://root:E5press0@${MONGO_HOST:localhost}:27017/madie?authSource=admin&maxPoolSize=50&connectTimeoutMS=2000&serverSelectionTimeoutMS=2000
```
Comment (sb-cecilialiu) [inline src/main/resources/application.yml]: Are we changing database as well?

Comment (jkotanchik-SB) [reply src/main/resources/application.yml]: Nice catch, I committed too many changes.

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -20,8 +20,7 @@ okta:
 
 client:
   vsac_base_url: https://vsac.nlm.nih.gov/vsac
-  service_ticket_endpoint: /ws/Ticket/%s?service=http://umlsks.nlm.nih.gov
-  valueset_endpoint: /svs/RetrieveMultipleValueSets?id={oid}&ticket={st}&profile={profile}&includeDraft={includeDraft}
+  valueset_endpoint: /svs/RetrieveMultipleValueSets?id={oid}&profile={profile}&includeDraft={includeDraft}
   default_profile: ${DEFAULT_VSAC_PROFILE:Most Recent Code System Versions in VSAC}
   uts_login: /ws/Ticket
```
Comment (adongare) [inline src/main/resources/application.yml]: this is probably not required anymore?

**Review by adongare — DISMISSED** [review]

File: src/main/resources/application.yml (hunk 2 above)
Comment (jkotanchik-SB) [reply src/main/resources/application.yml]: Agreed. Removed it.

**Review by adongare — APPROVED** [review]
